src/electron/platform/android/appium-service-configurator.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import ADB from 'appium-adb';
+import {
+    AndroidServiceConfigurator,
+    DeviceInfo,
+    PackageInfo,
+    PermissionInfo,
+} from 'electron/platform/android/android-service-configurator';
+
+type AdbDevice = {
+    udid: string;
+};
+
+const servicePackageName: string = 'com.microsoft.accessibilityinsightsforandroidservice';
+
+export class AppiumServiceConfigurator implements AndroidServiceConfigurator {
+    private readonly adb: ADB;
+
+    constructor(adb: ADB) {
+        this.adb = adb;
+    }
+
+    public getConnectedDevices = async (): Promise<Array<DeviceInfo>> => {
+        // First add devices that are flagged as emulator, then
+        // add devices that aren't flagged as emulators
+        const detectedDevices = {};
+        await this.addDevices(await this.adb.getConnectedEmulators(), true, detectedDevices);
+        await this.addDevices(await this.adb.getConnectedDevices(), false, detectedDevices);
+
+        const output: Array<DeviceInfo> = [];
+        Object.values(detectedDevices).forEach(device => output.push(device as DeviceInfo));
+        return output;
+    };
+
+    private async addDevices(
+        devices: Array<AdbDevice>,
+        isEmulator: boolean,
+        detectedDevices: any,
+    ): Promise<void> {
+        for (let loop = 0; loop < devices.length; loop++) {
+            const id = devices[loop].udid;
+            if (!detectedDevices[id]) {
+                const deviceInfo: DeviceInfo = await this.getDeviceInfo(id, isEmulator);
+                detectedDevices[id] = deviceInfo;
+            }
+        }
+    }
+
+    private async getDeviceInfo(id: string, isEmulator: boolean): Promise<DeviceInfo> {
+        this.adb.setDeviceId(id);
+        const friendlyName = await this.adb.getModel();
+        return { id, isEmulator, friendlyName };
+    }
+
+    public getPackageInfo = async (deviceId: string): Promise<PackageInfo> => {
+        this.adb.setDeviceId(deviceId);
+        const info: PackageInfo = await this.adb.getPackageInfo(servicePackageName);
+        const output: PackageInfo = {};
+        if (info?.versionCode) {
```

Comment:
As a rule, I don't recommend relying on how javascript decides whether a non-boolean is "truthy" or not, it leads to some surprising behavior a lot of the time (eg, as-written, if versionCode is 0 or versionName is an empty string, those will both propagate as `undefined` from this function)

I think you can avoid that issue and simplify the code by using this instead:

```ts
return {
    versionCode: info?.versionCode,
    versionName: info?.versionName,
};
```

Reply:
Done